### ISSUE TITLE
`keyword-spacing` and `type-annotation-spacing`: conflicts in v2.6.5

### ISSUE THREAD

**Issue by Cherry** [eslint-stylistic/eslint-stylistic#525]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Since v2.6.5, there are conflicts between the `keyword-spacing` and `type-annotation-spacing` rule without any config changes from prior versions.

- Open reproduction
- Run `npm run lint` - observe `@stylistic/keyword-spacing` error
- Add space before `as` in `index.ts`: `const foo = {bar: true} as const;`
- Run `npm run lint` - observe `@stylistic/type-annotation-spacing` error

This didn't previously happen in versions <v2.6.5. I imagine it's a result of https://github.com/eslint-stylistic/eslint-stylistic/pull/517. This can be worked around by changing the config of `type-annotation-spacing` to match `keyword-spacing` for operators, but this doesn't feel super ergonomic:
```js
'@stylistic/type-annotation-spacing': ['error', {
  before: false,
  after: true,
  overrides: {
    operator: {
      before: true,
      after: true
    }
  }
}],
```

### Reproduction

https://stackblitz.com/edit/github-yn7u8v?file=eslint.config.js

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
